scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,36 @@
+# 0005 - Adapt Scanner V4 NVD CVSS data updating pipeline to integrate with NVD CVE Api
+
+- **Author(s):** Yi Li <yli3@redhat.com>
+- **Created:** [2023-10-10]
+
+## Status
+
+Accepted
+
+## Context
+
+The National Vulnerability Database (NVD) has announced the retirement of its JSON feeds by December 15, 2023. This creates the need to change our existing CVSS data updater pipeline for Scanner V4, which currently relies on these JSON feeds. 
+
+The NVD JSON feed offers bundled data on a yearly basis, spanning from 2002 to the present year. A benefit is that only two requests are needed in the workflow for each year (one for the meta file and one for the data bundle). However, a limitation is the absence of filtering capabilities, such as by CVSS score or modification date. In contrast, the NVD API allows data retrieval through specific HTTP URL parameters, such as publish date, modified date, or CVSS score, enabling more precise data acquisition.
+
+## Decision
+
+The updated NVD CVSS GitHub Workflow (CI) has transitioned to sourcing data from the NVD CVE API, rather than the NVD Json feeds, ensuring the timely update of the NVD CVSS data bundle in Google Storage. To fetch data, the workflow downloads based on both the start index and CVSS v3 severity. For instance, a command such as curl "https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0" > critical-1.json would be used. For every severity level, the workflow iteratively exhaust the startIndex to capture all associated data. And there are four severity levels in total. After all json files downloaded, they will be compress to one data bundle as one single zip file.
```

Comment:
The pull by severity seems OK for now to filter out entries without a CVSS v3 (assuming thats the reason why). 

This seems as though it will limit scanner's ability to detect older / newer vulnerabilities.  I understand that ClairCore currently skips anything without a CVSS v3, I assume that will NOT the case forever.  Such as when CVSS v4 becomes widespread, this pull by severity may become more complex as we will have duplicates when pulling by CVSS v3 severities and also by CVSS v4 severities.  It may make more sense to pull all the vulns (irregardless of CVSS v3 severity) and relying on either Clairs or the CI job's logic to filter out the invalid entries in a single iteration vs. dealing with duplicates in feeds.  This will also give us a datasource to address issues that come up as to 'why wasn't CVE XYZ detected', we could take a look at the CI job and see that it was filtered due to a missing score.

Reply:
That's a valid point. I am not sure how soon CVSS V4 score will be widely adopted. What do you guys think? @jvdm @dcaravel

Reply:
I think when CVSS V4 scores are going to appear is not the main point, I think he's trying to ask you if grouping by severity will create any blockers in the future -- when it happens. And also, more importantly, is it just cheaper to get all vulns right now to make that future brighter and simpler? Since you're more familiar with the NVD APIs and the overall vuln updating pipeline, we would like to know if you have any thoughts about the advantages or disadvantages of pulling all vulns today or dealing with new score types later on.

Reply:
Even with the API key, retrieving all the data is a lengthy task. We're working to expedite the Consolidation in short term. It's hard to predict what will happen in the future and how much impact that future change would bring.
Also the question 'why wasn't CVE XYZ detected' probably won't be true, they probably would be wondering 'why CVE XYZ is missing a score'.

I think we can fetch by severity for now. 
We can always change our way of fetching data and add the filtering feature in our customized enricher (we will need it anyway to parse the json in scanner) after V4 metrics is out (or other situation happens). I feel that won't be too much of a change by that time.